sources/packages/api/src/route-controllers/student/student.controller.ts

```diff
@@ -579,4 +579,47 @@ export class StudentController extends BaseController {
       hasRestriction: studentRestrictionStatus.hasRestriction,
     } as StudentDetailDTO;
   }
+
+  /**
+   * controller to save the files submitted by the student
+   * with student uploader form.
+   * @Body payload
+   */
+  @AllowAuthorizedParty(AuthorizedParties.student)
+  @Post("upload-files")
+  async saveStudentUploadedFiles(
+    @UserToken() userToken: IUserToken,
+    @Body() payload: StudentFileUploader,
+  ): Promise<void> {
+    const existingStudent = await this.studentService.getStudentByUserId(
+      userToken.userId,
+    );
+    if (!existingStudent) {
+      throw new NotFoundException(
+        `No student was found with the student id ${userToken.userId}`,
```

Comment:
Please not expose the user id here since it is not an explicit part of the data sent by the user.
I would say that a simple "Student not found" would be enough.
We should have implement a guard to avoid these checks in the future. We have a tech debit already in place.

Reply:
I understand I copy-pasted this exception from another controller for the sake of consistency. I will update